### PR TITLE
Release v0.14.1

### PR DESCRIPTION
**Pull request by angela97lin**
# v0.14.1 Sep. 29, 2020

### Enhancements
 - Updated partial dependence methods to support calculating numeric columns in a dataset with non-numeric columns #1150
 - Added `get_feature_names` on `OneHotEncoder` #1193
 - Added `detect_problem_type` to `problem_type/utils.py` to automatically detect the problem type given targets #1194
 - Added LightGBM to AutoMLSearch #1199
 - Updates scikit-learn and scikit-optimize to use latest versions - 0.23.2 and 0.8.1 respectively #1141
 - Added `__str__` and `__repr__` for pipelines and components #1218
 - Included internal target check for both training and validation data in AutoMLSearch #1226
 - Add `ProblemTypes.all_problem_types` helper to get list of supported problem types #1219
 - Added `DecisionTreeClassifier` and `DecisionTreeRegressor` classes #1223
 - Added `ProblemTypes.all_problem_types` helper to get list of supported problem types #1219
 - `DataChecks` can now be parametrized by passing a list of `DataCheck` classes and a parameter dictionary #1167
 - Added first CV fold score as validation score in `AutoMLSearch.rankings` #1221
 - Updated `flake8 configuration to enable linting on `__init__.py` files #1234
 - Refined `make_pipeline_from_components` implementation #1204
### Fixes
 - Updated GitHub URL after migration to Alteryx GitHub org #1207
 - Changed Problem Type enum to be more similar to the string name #1208
 - Wrapped call to scikit-learn's partial dependence method in a `try`/`finally` block #1232
### Changes
 - Added `allow_writing_files` as a named argument to CatBoost estimators. #1202
 - Added `solver` and `multi_class` as named arguments to LogisticRegressionClassifier #1202
 - Replaced pipeline's `._transform` method to evaluate all the preprocessing steps of a pipeline with `.compute_estimator_features` #1231
 - Changed default large dataset train/test splitting behavior #1205
### Documentation Changes
 - Included description of how to access the component instances and features for pipeline user guide #1163
 - Updated API docs to refer to target as "target" instead of "labels" for non-classification tasks and minor docs cleanup #1160
 - Added Class Imbalance Data Check to `api_reference.rst` #1190 #1200
 - Added pipeline properties to API reference #1209
 - Clarified what the objective parameter in AutoML is used for in AutoML API reference and AutoML user guide #1222
 - Updated API docs to include `skopt.space.Categorical` option for component hyperparameter range definition #1228
 - Added install documentation for `libomp` in order to use LightGBM on Mac #1233
 - Improved description of `max_iterations` in documentation #1212
 - Removed unused code from sphinx conf #1235
###Testing Changes

### Breaking Changes
 - DefaultDataChecks now accepts a problem_type parameter that must be specified #1167
 - Pipeline's `._transform` method to evaluate all the preprocessing steps of a pipeline has been replaced with `.compute_estimator_features` #1231
 - `get_objectives` has been renamed to `get_core_objectives`. This function will now return a list of valid objective instances #1230

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -41,7 +49,7 @@ Release Notes
 .. warning::
 
     **Breaking Changes**
-        * DefaultDataChecks now accepts a problem_type parameter that must be specified :pr:`1167`
+        * `DefaultDataChecks` now accepts a `problem_type` parameter that must be specified :pr:`1167`
         * Pipeline's `._transform` method to evaluate all the preprocessing steps of a pipeline has been replaced with `.compute_estimator_features` :pr:`1231`
         * `get_objectives` has been renamed to `get_core_objectives`. This function will now return a list of valid objective instances :pr:`1230`
```
Comment (dsherry) [inline docs/source/release_notes.rst:54]: Nice this is a short list!

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -41,7 +49,7 @@ Release Notes
 .. warning::
 
     **Breaking Changes**
-        * DefaultDataChecks now accepts a problem_type parameter that must be specified :pr:`1167`
+        * `DefaultDataChecks` now accepts a `problem_type` parameter that must be specified :pr:`1167`
         * Pipeline's `._transform` method to evaluate all the preprocessing steps of a pipeline has been replaced with `.compute_estimator_features` :pr:`1231`
```
Comment (dsherry) [inline docs/source/release_notes.rst:53]: This isn't technically breaking since its a private API. But, its ok!

Comment (angela97lin) [reply docs/source/release_notes.rst:53]: Ahaha yeah, @freddyaboulton and I thought since this could have been used that it'd be good to make note of it 👀 I can remove it though!

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -2,6 +2,14 @@ Release Notes
...
 
 **Future Releases**
+    * Enhancements
+    * Fixes
+    * Changes
+    * Documentation Changes
+    * Testing Changes
+
+
+**v0.14.1 Sep. 29, 2020**
```
Comment (dsherry) [inline docs/source/release_notes.rst:12]: Less that 2 weeks have passed since 0.13.2 and we've already got over 30 merges! 🔥🔥🔥

**Review by dsherry — APPROVED** [review]
LGTM!

File: docs/source/release_notes.rst (hunk 2 above)
Comment (dsherry) [reply docs/source/release_notes.rst:53]: Yeah that's a good point, keep it!

**Comment by dsherry** [comment]
@angela97lin RTD builds are [still backed up](https://readthedocs.com/projects/feature-labs-inc-evalml/builds/). Let's wait to merge this until we verify the docs are good to go. Hopefully it won't be more than 60min 😬

**Review by freddyaboulton — APPROVED** [review]
Conda tests passed! https://github.com/conda-forge/evalml-core-feedstock/pull/8

**Comment by angela97lin** [comment]
Thanks guys! [Retriggering docs](https://readthedocs.com/projects/feature-labs-inc-evalml/builds/436195/) since the last one kept timing out, will merge when it's green and ready to go!
